libvast/vast/concept/parseable/vast/uuid.hpp

```diff
@@ -57,13 +54,15 @@ struct uuid_parser : parser<uuid_parser> {
           return false;
         c = *f++;
       }
-      byte = lookup(c);
+      if constexpr (std::is_same_v<Attribute, uuid>)
+        x[i] = lookup(c);
       if (f == l)
         return false;
       c = *f++;
-      byte <<= 4;
-      byte |= lookup(c);
-      ++i;
+      if constexpr (std::is_same_v<Attribute, uuid>) {
+        x[i] <<= 4;
```

Comment:
Could go in one line: `x[i] = (x[i] << 4) | lookup(c);`

Reply:
Good point, fixed.